INTMDB:273: Fix replication_specs update error

abner-dou opened the pull request:
## Description

When update `replication_specs` property, the provider tried to send an empty id, so this fix gets the saved id and send to the API on change of this property

Also a minor changed to the `acceptance_test` naming (from `acceptances_test` -> `acceptance_test`)

Link to any related issue(s):[INTMDB-237](https://jira.mongodb.org/browse/INTMDB-273) https://github.com/mongodb/terraform-provider-mongodbatlas/issues/588


## Type of change:

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [X] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
- [ ] Documentation fix/enhancement

## Required Checklist:

- [ ] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [ ] I have read the Terraform contribution guidelines
- [ ] I have added tests that prove my fix is effective or that my feature works per HashiCorp requirements
- [ ] I have added any necessary documentation (if appropriate)
- [X] I have run make fmt and formatted my code

## Further comments

ebostijancic committed:
fixes failing snapshots because timeout is too short (#583)

* fixes failing snapshots because timeout is too short

* uses default timeout from Default field

* improves documentation of the duration field

* fixes failing tests

abner-dou committed:
fix replication_specs update error

themantissa commented:
@abner-dou  can you please add the issue to PR description?  You put the internal ticket but not the cited github issue so this PR won't show up there.

thetonymaster commented on mongodbatlas/resource_mongodbatlas_cluster.go:
```diff
@@ -1179,10 +1180,8 @@ func expandReplicationSpecs(d *schema.ResourceData) ([]matlas.ReplicationSpec, e
-						id = oldSpecs["id"].(string)
-						break
-					}
+					id = oldSpecs["id"].(string)
```
if  `replication_specs` has only one element, shouldn't it have the max len set to `1` too?

abner-dou replied:
this resource can have multiple replication specs per cluster per zone

abner-dou committed:
fix

sonlexqt committed:
Fix typo: mongodbatlast => mongodbatlas (#589)

abner-dou committed:
INTMDB-270: Added  'with_default_alerts_settings' to project resource (#595)

* added  'with_default_alerts_settings' to project resource

* change project documentation

coderGo93 committed:
INTMDB-275: Changed the pointer in some paremeters for custom db role (#596)

* chore: updated vendor

* fix: fixes the pointer error for custom roles

* fixes a import failing for cloud provider snapshot

dependabot[bot] committed:
Chore(deps): Bump github.com/gruntwork-io/terratest (#599)

Bumps [github.com/gruntwork-io/terratest](https://github.com/gruntwork-io/terratest) from 0.38.2 to 0.38.4.
- [Release notes](https://github.com/gruntwork-io/terratest/releases)
- [Commits](https://github.com/gruntwork-io/terratest/compare/v0.38.2...v0.38.4)

---
updated-dependencies:
- dependency-name: github.com/gruntwork-io/terratest
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

abner-dou committed:
fix issue with project resource importer test (#601)

themantissa committed:
Update MDB version info (#600)

* Update MDB version info

Update resources for cluster, advanced cluster and custom roles to remove EoLed versions.

* Updates based on PR

jkodroff committed:
Fix docs for third party data source. (#604)

abner-dou committed:
Fix timeout in acctest (#602)

* fix timeout in serverless and private endpoint

* change timeout in testacc

* added default timeout

coderGo93 committed:
INTMDB-271: Fixing a bug and improving for custom zone mappings in global cluster config (#597)

* refactor: improved the funcionality of updating custom zone mapping, fixes error about duplicated

* test: added testacc

* refactor suggested by tony

* refactor

beergeek committed:
Add Organisation and Project invitations (#560)

* Add Organisation and Project invitations

* Fix linting issues

* Change to American spelling

* Apply fixes and suggested modifications

* Fix the linting issues

* Fix the linting issues

* Add tests

* Refactor portions of the code to reduce the amount of code required

* Changes as requested

* Changes as requested

* Changes to code as requested, plus doc updates

* Fix conflict

* Fix context and spelling

* Fix spelling and grammar as advised

* Further spelling and grammatical fixes

* refactor: fixed many bugs that couldn't be tested locally

Co-authored-by: beergeek <beergeek@beergeek.nakasendo.local>
Co-authored-by: Melissa Plunkett <themantissa@users.noreply.github.com>
Co-authored-by: Edgar López <edgars.ealo@gmail.com>

abner-dou commented:
Closed PR due is outdated
